Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/EngineView.cpp

```diff
@@ -163,12 +237,279 @@ namespace winrt::BabylonReactNative::implementation {
+            D3D_FEATURE_LEVEL_11_0,
+            D3D_FEATURE_LEVEL_10_1,
+            D3D_FEATURE_LEVEL_10_0,
+            D3D_FEATURE_LEVEL_9_3,
+            D3D_FEATURE_LEVEL_9_2,
+            D3D_FEATURE_LEVEL_9_1
+        };
+
+        ID3D11Device* device{ nullptr };
+        ::Microsoft::WRL::ComPtr<ID3D11DeviceContext> context;
+        D3D11CreateDevice(
+            nullptr,
+            D3D_DRIVER_TYPE_HARDWARE,
+            0,
+            D3D11_CREATE_DEVICE_BGRA_SUPPORT
+#if defined(_DEBUG)
+            | D3D11_CREATE_DEVICE_DEBUG
+#endif
+            ,
+            featureLevels,
+            ARRAYSIZE(featureLevels),
+            D3D11_SDK_VERSION,
+            &device,
+            NULL,
+            &context
+        );
+
+        // Get D3D11.1 device
+        if (s_d3dDevice) s_d3dDevice->Release();
+        device->QueryInterface(__uuidof(ID3D11Device1), (void**)&s_d3dDevice);
+
+        // Get D3D11.1 context
+        context.As(&s_d3dContext);
+
+        // Check MultiSampleQuality
+        for (uint32_t i = 1, last = 0; i < s_msaa.size(); ++i)
+        {
+            uint32_t msaa = s_checkMsaa[i];
+            uint32_t quality = 0;
+            HRESULT hr = s_d3dDevice->CheckMultisampleQualityLevels(DXGI_FORMAT_B8G8R8A8_UNORM, msaa, &quality);
+
+            if (SUCCEEDED(hr) && 0 < quality)
+            {
+                s_msaa[i].Count = msaa;
+                s_msaa[i].Quality = quality - 1;
+                last = i;
+            }
+            else
+            {
+                s_msaa[i] = s_msaa[last];
+            }
+        }
+    }
+
+    void EngineView::CreateSizeDependentResources()
+    {
+        s_d3dContext->OMSetRenderTargets(0, nullptr, nullptr);
+        s_d3dContext->Flush();
+
+        // Set render target size to the rendered size of the panel including the composition scale, 
+        // defaulting to the minimum of 1px if no size was specified.
+        _renderTargetWidth = _width * _compositionScaleX;
+        _renderTargetHeight = _height * _compositionScaleY;
+
+        // If the swap chain already exists, then resize it.
+        if (_swapChain)
+        {
+            _backBufferPtr = nullptr;
+            _depthBufferPtr = nullptr;
+
+            HRESULT hr = _swapChain->ResizeBuffers(
+                2,
+                static_cast<UINT>(_renderTargetWidth),
+                static_cast<UINT>(_renderTargetHeight),
+                DXGI_FORMAT_B8G8R8A8_UNORM,
+                0
+            );
+
+            if (hr == DXGI_ERROR_DEVICE_REMOVED || hr == DXGI_ERROR_DEVICE_RESET)
+            {
+                // If the device was removed for any reason, a new device and swap chain will need to be created.
+                OnDeviceLost();
+                return;
+
+            }
+        }
+        else // Otherwise, create a new one
+        {
+            _sampleDesc = s_msaa[static_cast<uint32_t>(MSAA::X4)];  // MSAA::X4 Default Setting in BabylonNative

```

Comment:
Seems like this is fine for now, but it might be worth filing an issue to eventually make this externally configurable, since I could see BNRN developers potentially wanting control over the MSAA level to balance performance / video memory / visual quality.

Reply:
that's right